### DISCUSSION TITLE
How to calculate HEI2020 from 2017 to March 2020  in NHANSE？

### DISCUSSION THREAD

**Discussion by ZAJ0930** [jamesjiadazhan/dietaryindex#236] [Q&A] [answered]
When I run the code with the data of NHANSE2017-2018, it can works. 
When I run the code with the data of NHANSE2017-2020, it can `t works. 
FPED_PATH_2 = "~/Desktop/data/liver/data/fped_dr1tot_1720.sas7bdat"
> NUTRIENT_PATH_2 = "~/Desktop/data/liver/data/P_DR1TOT.XPT"
> DEMO_PATH_2 = "~/Desktop/data/liver/data/P_DEMO.XPT"
> HEI2020_NHANES_FPED(FPED_PATH=FPED_PATH_2, NUTRIENT_PATH=NUTRIENT_PATH_2, DEMO_PATH=DEMO_PATH_2)
Error in `dplyr::select()`:
! Can't subset columns that don't exist.
✖ Column `WTDRD1` doesn't exist.
Run `rlang::last_trace()` to see where the error occurred.
> rlang::last_trace()
<error/vctrs_error_subscript_oob>
Error in `dplyr::select()`:
! Can't subset columns that don't exist.
✖ Column `WTDRD1` doesn't exist.
---
Backtrace:
    ▆
 1. ├─dietaryindex::HEI2020_NHANES_FPED(...)
 2. │ └─... %>% arrange(SEQN)
 3. ├─dplyr::arrange(., SEQN)
 4. ├─dplyr::select(...)
 5. └─dplyr:::select.data.frame(...)
Run rlang::last_trace(drop = FALSE) to see 18 hidden frames.
> rlang::last_trace(drop = FALSE)
<error/vctrs_error_subscript_oob>
Error in `dplyr::select()`:
! Can't subset columns that don't exist.
✖ Column `WTDRD1` doesn't exist.
---
Backtrace:
     ▆
  1. ├─dietaryindex::HEI2020_NHANES_FPED(...)
  2. │ └─... %>% arrange(SEQN)
  3. ├─dplyr::arrange(., SEQN)
  4. ├─dplyr::select(...)
  5. ├─dplyr:::select.data.frame(...)
  6. │ └─tidyselect::eval_select(expr(c(...)), data = .data, error_call = error_call)
  7. │   └─tidyselect:::eval_select_impl(...)
  8. │     ├─tidyselect:::with_subscript_errors(...)
  9. │     │ └─rlang::try_fetch(...)
 10. │     │   └─base::withCallingHandlers(...)
 11. │     └─tidyselect:::vars_select_eval(...)
 12. │       └─tidyselect:::walk_data_tree(expr, data_mask, context_mask)
 13. │         └─tidyselect:::eval_c(expr, data_mask, context_mask)
 14. │           └─tidyselect:::reduce_sels(node, data_mask, context_mask, init = init)
 15. │             └─tidyselect:::walk_data_tree(new, data_mask, context_mask)
 16. │               └─tidyselect:::as_indices_sel_impl(...)
 17. │                 └─tidyselect:::as_indices_impl(...)
 18. │                   └─tidyselect:::chr_as_locations(x, vars, call = call, arg = arg)
 19. │                     └─vctrs::vec_as_location(...)
 20. └─vctrs (local) `<fn>`()
 21.   └─vctrs:::stop_subscript_oob(...)
 22.     └─vctrs:::stop_subscript(...)
 23.       └─rlang::abort(...)
Please help me, thanks.

**Comment by jamesjiadazhan** [comment, marked as answer]
I tried to replicate your error, but it seems that both using the internal dataset and external downloaded dataset work from my end.

Please download the dietaryindex complied NHANES dataset here: https://github.com/jamesjiadazhan/dietaryindex_NHANES/tree/main/data/NHANES_combined

Then, use the following codes. It should work (change your setwd path accordingly):
```
setwd("/Users/james/Desktop/Emory University - Ph.D./dietaryindex_package/NHANES_combined/dietaryindex_NHANES/data/NHANES_combined")
load("NHANES_20172020.Rda")

HEI2020_NHANES_FPED(
    FPED_PATH = NHANES_20172020$FPED, 
    NUTRIENT_PATH = NHANES_20172020$NUTRIENT, 
    DEMO_PATH = NHANES_20172020$DEMO)
```
